src/util/config.js

```diff
@@ -1,83 +1,153 @@
-import Joi from 'joi';
+import fs from 'fs';
+import convict from 'convict';
+
 import {
 	duotones,
 	getDuotoneUrls
 } from './duotone';
+
 /**
- * Start the server with a config
- *
- * This module provides a Promise-based interface for assembling a
- * configuration object and passing it to the server startup function.
  * @module config
  */
 
-export default function getConfig(overrideConfig) {
-	/**
-	 * Read all config from environment variables here once on startup
-	 */
-	if (!process.env.OAUTH_ACCESS_URL && process.env.ANONYMOUS_ACCESS_URL) {
-		console.warn('The ANONYMOUS_ACCESS_URL env variable is deprecated - please rename to OAUTH_ACCESS_URL');
-	}
-	if (!process.env.OAUTH_AUTH_URL && process.env.ANONYMOUS_AUTH_URL) {
-		console.warn('The ANONYMOUS_AUTH_URL env variable is deprecated - please rename to OAUTH_AUTH_URL');
-	}
-	const config = {
-		API_PROTOCOL: process.env.API_PROTOCOL || 'https',
-		API_HOST: process.env.API_HOST || 'api.dev.meetup.com',
-		API_TIMEOUT: process.env.API_TIMEOUT || 8000,
-		COOKIE_ENCRYPT_SECRET: process.env.COOKIE_ENCRYPT_SECRET,
-		CSRF_SECRET: process.env.CSRF_SECRET,
-		DEV_SERVER_PORT: process.env.DEV_SERVER_PORT || 8000,
-		OAUTH_AUTH_URL: process.env.OAUTH_AUTH_URL ||
-			process.env.ANONYMOUS_AUTH_URL ||
-			'https://secure.dev.meetup.com/oauth2/authorize',
-		OAUTH_ACCESS_URL: process.env.OAUTH_ACCESS_URL ||
-			process.env.ANONYMOUS_ACCESS_URL ||
-			'https://secure.dev.meetup.com/oauth2/access',
-		PHOTO_SCALER_SALT: process.env.PHOTO_SCALER_SALT,
-		oauth: {
-			secret: process.env.MUPWEB_OAUTH_SECRET,
-			key: process.env.MUPWEB_OAUTH_KEY,
+const oauthError = new Error('get oauth secrets from #web-platform team');
+
+let config = convict({
```

Comment:
I'm considering naming this `envConfig` so it doesn't get confused w/ every other configuration in the server code (there are a few). Thoughts?